### PR TITLE
Kyma labels validation and function controller update 

### PR DESCRIPTION
**Pull request by MichalKalke**
**Description**

Changes proposed in this pull request:

- Function controller uses workload's metadata from new fields in function CRD `spec.templates.function|build.metadata`
- Validation webhook shouldn't allow to add labels from `serverless.kyma-project.io` group

**Related issue(s)**
Related to https://github.com/kyma-project/kyma/issues/14877

### CODE REVIEW

File: components/function-controller/internal/controllers/serverless/build_resources.go [hunk 1]
```diff
@@ -119,10 +119,10 @@ func (r *FunctionReconciler) deploymentSelectorLabels(instance *serverlessv1alph
 }
 
 func (r *FunctionReconciler) podLabels(instance *serverlessv1alpha2.Function) map[string]string {
```
Comment (anoipm) [inline components/function-controller/internal/controllers/serverless/build_resources.go:121]: This function is unused - is used only by test

File: components/function-controller/internal/controllers/serverless/build_resources_test.go [hunk 2]
```diff
@@ -334,14 +337,17 @@ func TestFunctionReconciler_servicePodLabels(t *testing.T) {
 				UID:  "fn-uuid",
 			},
 				Spec: serverlessv1alpha2.FunctionSpec{
-					Template: &serverlessv1alpha2.Template{
-						Labels: map[string]string{
-							"test-some":                              "test-label",
-							serverlessv1alpha2.FunctionResourceLabel: "job",
-							serverlessv1alpha2.FunctionNameLabel:     "some-other-name",
+					Templates: &serverlessv1alpha2.Templates{
```
Comment (anoipm) [inline components/function-controller/internal/controllers/serverless/build_resources_test.go:340]: This test use function unused by others (except this test).

File: components/function-controller/pkg/apis/serverless/v1alpha2/function_validation.go [hunk 3]
```diff
@@ -324,11 +324,30 @@ func (spec *FunctionSpec) validateLabels(_ *ValidationConfig) error {
...
+	}
 
 	errs := v1validation.ValidateLabels(labels, fieldPath)
 	return errs.ToAggregate()
 }
 
+func validateKymaLabels(labels map[string]string) error {
+	allErrs := []string{}
+	for key, _ := range labels {
+		isUsed, err := regexp.MatchString(".*serverless.kyma-project.io*.", key)
```
[the remark this answers is not in the document]
Comment (MichalKalke) [reply components/function-controller/pkg/apis/serverless/v1alpha2/function_validation.go]: Fixed.

**Comment by anoipm** [comment]
Should we document these changes?

**Review by anoipm — CHANGES_REQUESTED** [review]
Should we update documentation connected these changes?

File: components/function-controller/pkg/apis/serverless/v1alpha2/function_validation.go (hunk 3 above)
[the remark this answers is not in the document]
Comment (dbadura) [reply components/function-controller/pkg/apis/serverless/v1alpha2/function_validation.go]: Please don't use regexp, `startwith` is enought for this use case, eg.: `serverless.kyma.project.io/`

File: components/function-controller/internal/controllers/serverless/build_resources.go (hunk 1 above)
Comment (MichalKalke) [reply components/function-controller/internal/controllers/serverless/build_resources.go]: Removed unused functions.

**Comment by MichalKalke** [comment]
> Should we update documentation connected these changes?
@anoipm Could you create an issue for our TW?

File: components/function-controller/pkg/apis/serverless/v1alpha2/function_validation.go (hunk 3 above)
[the remark this answers is not in the document]
Comment (MichalKalke) [reply components/function-controller/pkg/apis/serverless/v1alpha2/function_validation.go]: Changed to `strings.hasPrefix()`

**Pull request closed without merging**